File: Pathfinder_1_13_1/src/main/java/com/github/ysl3000/bukkit/pathfinding/craftbukkit/v1_13_R2/pathfinding/CraftPathfinderPlayer.java
package com.github.ysl3000.bukkit.pathfinding.craftbukkit.v1_13_R2.pathfinding;

import org.bukkit.craftbukkit.v1_13_R2.entity.CraftPlayer;
import org.bukkit.entity.Player;

import com.github.ysl3000.bukkit.pathfinding.pathfinding.PathfinderPlayer;

import net.minecraft.server.v1_13_R2.EntityPlayer;

/**
 * Created by ysl3000
 */
public class CraftPathfinderPlayer implements PathfinderPlayer {

  public final double STILL = -0.0784000015258789;

  private EntityPlayer entityPlayer;

  public CraftPathfinderPlayer(Player player) {
    this(((CraftPlayer) player).getHandle());
  }

  private CraftPathfinderPlayer(EntityPlayer entityPlayer) {
    this.entityPlayer = entityPlayer;
  }

  @Override
  public Player getPlayer() {
    return entityPlayer.getBukkitEntity();
  }

  @Override
  public double getRelativeMotionX() {
    return entityPlayer.motX;
  }

  @Override
  public double getRelativeMotionY() {
    return entityPlayer.motY;
  }

  @Override
  public double getRelativeMotionZ() {
    return entityPlayer.motZ;
  }

  @Override
  public float getRelativeMotionYaw() {
    return entityPlayer.yaw;
  }

  @Override
  public float getRelativeMotionPitch() {
    return entityPlayer.pitch;
  }

  @Override
  public float getMotionForward() {
    return entityPlayer.bh;
  }

  @Override
  public float getMotionSideward() {
    return entityPlayer.bj;
  }

  @Override
  public boolean isJumping() {
    return entityPlayer.motY > STILL;
  }


}

File: Pathfinder_1_13_1/src/main/java/com/github/ysl3000/bukkit/pathfinding/craftbukkit/v1_13_R2/pathfinding/CraftPathfinderPlayer.kt
package com.github.ysl3000.bukkit.pathfinding.craftbukkit.v1_13_R2.pathfinding

import com.github.ysl3000.bukkit.pathfinding.AbstractPathfinderPlayer
import net.minecraft.server.v1_13_R2.EntityPlayer
import org.bukkit.craftbukkit.v1_13_R2.entity.CraftPlayer
import org.bukkit.entity.Player

/**
 * Created by ysl3000
 */
class CraftPathfinderPlayer private constructor(ep: EntityPlayer) :
        AbstractPathfinderPlayer(
                player = { ep.bukkitEntity },
                relativeMotionX = { ep.motX },
                relativeMotionY = { ep.motY },
                relativeMotionZ = { ep.motZ },
                relativeMotionYaw = { ep.yaw },
                relativeMotionPitch = { ep.pitch },
                relativeMotionForward = { ep.bh },
                relativeMotionSideward = { ep.bj },
                jump = { ep.motY > STILL }
        ) {
    constructor(player: Player) : this((player as CraftPlayer).handle)

    companion object {
        private const val STILL = -0.0784000015258789
    }
}
